### PR TITLE
load aws external access roles and role last used info

### PR DESCRIPTION
**Pull request by ganeshjagdale9545**

### CODE REVIEW

File: cartography/intel/aws/iam.py [hunk 1]
```diff
@@ -248,9 +248,32 @@ def get_role_list_data(boto3_session: boto3.session.Session) -> Dict:
...
+@timeit
+def get_external_access_roles(boto3_session: boto3.session.Session) -> List[Dict]:
+    try:
+        analyzer_client = boto3_session.client('accessanalyzer')
+        # AWS allow to create only one analyzer per account per region with type (ACCOUNT, ORGANIZATION, ACCOUNT_UNUSED_ACCESS, and ORGANIZTAION_UNUSED_ACCESS) thats why uses analyzers_list index 0
+        analyzers_list = analyzer_client.list_analyzers(type='ACCOUNT').get("analyzers", [])
+        findings: List = []
+        if analyzers_list:
+            paginator = analyzer_client.get_paginator('list_findings_v2')
+            for page in paginator.paginate(analyzerArn=analyzers_list[0].get("arn"), filter={'resourceType': {'eq': ['AWS::IAM::Role']}}):
```
Comment (mpurusottamc) [inline cartography/intel/aws/iam.py:266]: this analyzer query does not mention anything about external access role. how do we find that out? if there are any documents, can you please share?

File: cartography/intel/aws/iam.py [hunk 2]
```diff
@@ -248,9 +248,32 @@ def get_role_list_data(boto3_session: boto3.session.Session) -> Dict:
...
+                findings.extend(page.get("findings", []))
+            return findings
+        else:
+            analyzer_client.create_analyzer(analyzerName="cdx_analyzer", type="ACCOUNT", tags={'owner': 'cdx_lab', 'project': 'iam-entitlement'})
```
Comment (mpurusottamc) [inline cartography/intel/aws/iam.py:270]: let's change the tag values.
owner should be cloudanix
project - iam-entitlements-analyzer

**Review by mpurusottamc — CHANGES_REQUESTED** [review]
need some clarification and fixes

File: cartography/intel/aws/iam.py (hunk 1 above)
Comment (ganeshjagdale9545) [reply cartography/intel/aws/iam.py:266]: we getting findings for ACCOUNT type access analyzer which return the external access roles

**Review by mpurusottamc — APPROVED** [review]
looks good.

**Pull request merged**